docs/denoising.rst

```diff
@@ -0,0 +1,250 @@
+
+  The first thing to know is that aggressive denoising, while appropriate for orthogonal time series,
+  may remove signal associated with "good" time series that are independent to, but not orthogonal with,
+  the "bad" time series being regressed out.
+
+  As an alternative, users may want to try non-aggressive denoising or aggressive denoising combined with component orthogonalization.
+
+  The main difference between the orthogonalization+aggressive and non-aggressive approaches is that,
+  with orthogonalization,
+  all the variance common across the accepted and rejected components is put in the accepted basket,
+  and therefore it is not removed in the nuisance regression.
+  In contrast, with the non-aggressive denoising,
+  the shared variance is smartly split during the estimation process and therefore some amount is removed.
+
+Now we can get started!
+This walkthrough will show you how to perform different kinds of denoising on example data.
+
+For this walkthrough, we will use Python and the Nilearn package to denoised data.
+However, you can definitely accomplish all of these steps in other languages, like MATLAB,
+or using other neuroimaging toolboxes, like AFNI.
+
+.. important::
+
+  In this walkthrough, we show how to denoise outputs of the ``fMRIPrep`` preprocessing workflow.
+
+  As such, the filenames we use reflect the current standard for ``fMRIPrep`` outputs,
+  although you will need to adjust them based on your own data.
+  Additionally, these examples show how to incorporate external regressors
+  (in this case, nuisance regressors like motion parameters) into your denoising step.
+
+  If you do not want to include external regressors, you will need to adapt the code slightly;
+  however, the necessary changes should be minimal.
+
+Let's start by loading the necessary data.
+No matter which type of denoising you want to use, you will need to include this step.
+
+For this, you need the mixing matrix, the data you're denoising, a brain mask,
+and an index of "bad" components.
+
+.. code-block:: python
+
+  import pandas as pd  # A library for working with tabular data
+
+  # Files from fMRIPrep
+  data_file = "sub-01_task-rest_space-MNI152NLin2009cAsym_desc-preproc_bold.nii.gz"
+  mask_file = "sub-01_task-rest_desc-brain_mask.nii.gz"
+  confounds_file = "sub-01_task-rest_desc-confounds_timeseries.tsv"
+
+  # Files from tedana (after running on fMRIPrepped data)
+  mixing_file = "sub-01_task-rest_desc-ICA_mixing.tsv"
+  metrics_file = "sub-01_task-rest_desc-tedana_metrics.tsv"
+
+  # Load the mixing matrix
+  mixing_df = pd.read_table(mixing_file)  # Shape is time-by-components
+
+  # Load the component table
+  metrics_df = pd.read_table(metrics_file)
+  rejected_columns = metrics_df.loc[metrics_df["classification"] == "rejected", "Component"]
+  accepted_columns = metrics_df.loc[metrics_df["classification"] == "accepted", "Component"]
```

Comment:
This is one thing that confuses me. Do we _want_ to orthogonalize our rejected components with respect to a bunch of "accepted" components we aren't actually confident about? We keep the (currently) ignored components in because we don't have anything better to do with them, given the amount of variance they explain or how unsure the decision tree is about whether they're BOLD or non-BOLD. I don't think that means I'm confident enough to say "these are definitely BOLD and I don't want my rejected components to share any variance with them".

Reply:
The distinction between "accepted" and "ignored" is really fuzzy. In the current code, everything that isn't classified as rejected or ignored ends up in accepted. There are multiple reasons something might end up in ignored and I'm not sure I'd call all the actual criteria edge cases or things we are less confident about than the "accepted" components. This is one of several reasons the current terminology is problematic & "ignored" is being replaced with "accepted" plus more descriptive tags in the modularized code.

Reply:
I'm still wary of orthogonalizing rejected components on ones we're not sure about (or ones that don't explain a useful level of variance), but I also agree that the "ignored" label doesn't necessarily mean low-variance or ambiguous-classification, so I'll just leave as `== "accepted"`. Thanks!